spec/controllers/prompts_controller_spec.rb

```diff
@@ -0,0 +1,131 @@
+      #signups.collection.collection_preference.closed = true
+      post :create, collection_id: signups.collection.name 
+      expect(response).to redirect_to(collection_path(signups.collection))
+      expect(flash[:error]).to eq "Signup is currently closed: please contact a moderator for help."
+    end
+ end
+
+  describe 'not_signup_owner' do
+    it 'should show an error and redirect' do
+      signups = create(:challenge_signup)
+      prompt = signups.collection.prompts.first
+      post :edit, id: prompt.id, collection_id: signups.collection.name
+      expect(response).to redirect_to(collection_path(signups.collection))
+      expect(flash[:error]).to eq "You can't edit someone else's sign-up!"
+    end
+  end
+
+  describe 'new_prompt_offer' do
+    it 'should have no errors and redirect' do
+      signups = create(:challenge_signup)
+      signups.collection.challenge.signup_open = true
+      signups.collection.challenge.save
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      post :new, collection_id: signups.collection.name, prompt_type: 'offer'
+      expect(response).to have_http_status(200)
+      expect(flash[:error]).blank?
+    end
+  end
+
+  describe 'create_prompt_offer' do
+    it 'should have no errors and redirect' do
+      signups = create(:challenge_signup)
+      prompt = signups.collection.prompts.first
+      signups.collection.challenge.signup_open = true
+      signups.collection.challenge.save
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      post :create, collection_id: signups.collection.name, prompt_type: 'offer', prompt: {}
+      expect(response).to redirect_to("#{collection_signups_path(signups.collection)}/#{prompt.challenge_signup_id}/edit")
+      expect(flash[:notice]).blank?
+      expect(flash[:error]).blank?
+    end
+  end
+
+  describe 'update_prompt' do
+    it 'should have no errors and redirect' do
+      signups = create(:challenge_signup)
+      prompt = signups.collection.prompts.first
+      signups.collection.challenge.signup_open = true
+      signups.collection.challenge.save
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signups.collection).first.pseud_id).user)
+      put :update, collection_id: signups.collection.name, prompt_type: 'offer', prompt: {description: 'This is a description'}, id: prompt.id
+      expect(flash[:notice]).to eq 'Prompt was successfully updated.'
+      expect(flash[:error]).blank?
+      expect(response).to redirect_to("#{collection_signups_path(signups.collection)}/#{signups.id}")
+      new_prompt = signups.collection.prompts.first
+      expect(new_prompt.description).to eq("<p>This is a description</p>")
+   end
+  end
+
+  describe 'destroy' do
```

Comment:
Description-wise, maybe something more like:
```
describe "destroy" do
  it "redirects with an error when sign-ups are closed" do
  end
  it "redirects with an error when it would make a sign-up invalid" do
  end
  it "deletes the prompt and redirects with a success message" do
  end
end
```